plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/TrinoFileSystemCache.java

```diff
@@ -344,6 +351,58 @@ public FSDataOutputStream createNonRecursive(Path f, FsPermission permission, En
         }
     }
 
+    private static class LocalFileSystemWrapper
```

Comment:
How different is this from `FileSystemWrapper` ?

Reply:
java.lang.ClassCastException: class io.trino.plugin.hive.fs.TrinoFileSystemCache$FileSystemWrapper cannot be cast to class org.apache.hadoop.fs.LocalFileSystem (io.trino.plugin.hive.fs.TrinoFileSystemCache$FileSystemWrapper and org.apache.hadoop.fs.LocalFileSystem are in unnamed module of loader 'app')

	at org.apache.hadoop.fs.FileSystem.getLocal(FileSystem.java:435)

Reply:
Instead can we handle them in `TrinoFileSystemCache#createFileSystem` where we could directly create a `LocalFileSystem` instead of implementing a `LocalFileSystemWrapper`?